src/util/macros.h

```diff
@@ -14,7 +14,7 @@
 // classes.
 #define PROPERTY_SET_BYVAL_GET_BYREF(TYPE, NAME, CAP_NAME) \
 public: void set##CAP_NAME(TYPE NAME) { m_##NAME = std::move(NAME); } \
-public: TYPE const& get##CAP_NAME() const { return m_##NAME; } \
-public: TYPE& ref##CAP_NAME() { return m_##NAME; } \
+public: constexpr TYPE const& get##CAP_NAME() const { return m_##NAME; } \
```

Comment:
Are your referring to `TYPE const&` vs. `const TYPE&`? Both types are equivalent.

Reply:
For my understanding not.
In one case the referenced itself is const. In the other case the referenced object is const.

Reply:
> const applies to the thing left of it. If there is nothing on the left then it applies to the thing right of it.

Left-side const is the traditional version while right-side const is more consistent. Try it with a pointer or function type and you will notice why.

I will change it to avoid further discussions once again. This code has once been imported from a personal project.